### PR TITLE
feat: add the remote-build command

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -62,7 +62,7 @@ colorama==0.4.6
     # via sphinx-autobuild
 coverage==7.6.8
     # via rockcraft (pyproject.toml)
-craft-application==4.5.0
+craft-application @ git+https://github.com/canonical/craft-application@work/CRAFT-3748-remote-build-rocks
```
Comment (tigarmo) [inline requirements-dev.txt:65]: reminder to update these

File: .github/workflows/spread-manual.yaml [hunk 2]
```diff
@@ -43,7 +43,13 @@ jobs:
         with:
           name: snap
           path: tests
-      - name: Run spread (large/flaky)
+
+      - name: Run remote-build tests
+        env:
+          LAUNCHPAD_TOKEN: "${{ secrets.LAUNCHPAD_TOKEN }}"
```
Comment (cjdcordeiro) [inline .github/workflows/spread-manual.yaml:49]: Note that this secret won't be available for fork-based PRs ([ref](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflows-in-forked-repositories))

Comment (cjdcordeiro) [reply .github/workflows/spread-manual.yaml:49]: (i wouldn't make a big deal out of it though because this also runs on a schedule)

Comment (tigarmo) [reply .github/workflows/spread-manual.yaml:49]: yeah that's fine. After the feature stabilizes I don't expect this to break often (well, not from changes on our side)
also that token is a little too powerful so the fact that it's not available to forks is a good thing

File: rockcraft/cli.py [hunk 3]
```diff
@@ -54,4 +56,6 @@ def _create_app() -> "Rockcraft":
         ],
     )
 
+    app.add_command_group("Lifecycle", [appcommands.RemoteBuild])
```
Comment (mr-cal) [inline rockcraft/cli.py:59]: Nice to see craft-app's command group merging is working.

File: tests/spread/rockcraft/remote-build/task.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    echo "No credentials set in env LAUNCHPAD_TOKEN"
+    exit 1
+  fi
+
+  mkdir work
+  cd work
+  rockcraft init --name="remote-build"
+  
+  # commit the project
+  #git config --global --add safe.directory "$PWD"
```
Comment (mr-cal) [inline tests/spread/rockcraft/remote-build/task.yaml:19]: dead code?

**Review by mr-cal — APPROVED** [review]

File: rockcraft/cli.py (hunk 3 above)
Comment (tigarmo) [reply rockcraft/cli.py:59]: yeap thanks for this! Very useful

File: .github/workflows/spread-manual.yaml (hunk 2 above)
Comment (sergiusens) [reply .github/workflows/spread-manual.yaml:49]: these tests are also not triggered with every PR

**Review by sergiusens — APPROVED** [review]

**Pull request merged**